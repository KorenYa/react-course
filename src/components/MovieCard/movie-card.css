.movie-card {
    padding: 20px;
    color: #000000;
}

.movie-card__title {
    font-size: 14px;
    padding-top: 10px;
    text-transform: uppercase;
}

.movie-card__date {
    font-size: 10px;
    float: right;
    padding: 4px 6px;
    border: 1px solid #000000;
    border-radius: 2px;
}

.movie-card__genre {
    font-size: 10px;
    padding-top: 15px;
}