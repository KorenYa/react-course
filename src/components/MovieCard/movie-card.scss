@import '../../styles/constants.scss';

.movie-details {
    display: inline-block;
    padding: 20px;
    color: $white;

    &__title {
        font-size: $lg;
        font-weight: $bold;
        color: $red;
        padding-bottom: 10px;
        text-transform: uppercase;
    }

    &__tagline {
        font-size: $md;
        font-weight: $thin;
        padding: 10px 0;
    }

    &__date,
    &__runtime {
        font-weight: $bold;
        padding-right: 15px;
    }

    &__overview {
        font-size: $md;
        font-weight: $thin;
        padding-top: 20px;
    }
}
