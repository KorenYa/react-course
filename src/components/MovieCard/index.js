import React, { Component } from 'react';
import Image from '../Image';

import './movie-card.scss';
class MovieCard extends Component {
    render() {
        return (
            <div className='movie-card'>
                <Image src={this.props.poster} alt={this.props.title} />
                <h2 className='movie-card__title'>
                    {this.props.title}
                    <span className='movie-card__date'>{this.props.date}</span>
                </h2>
                <div className='movie-card__genre'>{this.props.genre}</div>
            </div>
        );
    }
}
export default MovieCard;
