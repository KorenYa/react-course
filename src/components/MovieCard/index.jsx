import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import Image from '../Image';
import './movie-card.scss';

const MovieCard = (props) => {
  const {
    id, poster, title, date, genre,
  } = props;
  return (
    <Link to={`/film/${id}`}>
      <div className="movie-card">
        <div className="text-center">
          <Image src={poster} alt={title} />
        </div>
        <h2 className="movie-card__title">
          {title}
          <span className="movie-card__date">{date}</span>
        </h2>
        <div className="movie-card__genre">{genre}</div>
      </div>
    </Link>
  );
};
MovieCard.propTypes = {
  id: PropTypes.number,
  poster: PropTypes.string,
  title: PropTypes.string,
  date: PropTypes.string,
  genre: PropTypes.string,
};
export default MovieCard;
