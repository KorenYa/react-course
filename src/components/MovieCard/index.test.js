import React from 'react';
import renderer from 'react-test-renderer';

import MovieCard from '../MovieCard';

describe('MovieCard', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(
                <MovieCard
                    poster='poster.jpg'
                    title='Movie'
                    date='27-03-2019'
                    genre='action'
                />
            )
            .toJSON();

        expect(tree).toMatchSnapshot();
    });
});
