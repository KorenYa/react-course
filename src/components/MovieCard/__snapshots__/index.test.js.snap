// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders correctly 1`] = `
<a
  href="/film/undefined"
  onClick={[Function]}
>
  <div
    className="movie-card"
  >
    <div
      className="text-center"
    >
      <img
        alt="Movie"
        className="img-responsive"
        src="poster.jpg"
      />
    </div>
    <h2
      className="movie-card__title"
    >
      Movie
      <span
        className="movie-card__date"
      >
        27-03-2019
      </span>
    </h2>
    <div
      className="movie-card__genre"
    >
      action
    </div>
  </div>
</a>
`;
